symengine/functions.cpp

```diff
@@ -2895,6 +2895,92 @@ RCP<const Basic> uppergamma(const RCP<const Basic> &s, const RCP<const Basic> &x
 }
 
 
+PolyGamma::PolyGamma(const RCP<const Basic> &n, const RCP<const Basic> &x)
+    : n_{n}, x_{x}
+{
+    SYMENGINE_ASSERT(is_canonical(n_, x_))
+}
+
+bool PolyGamma::is_canonical(const RCP<const Basic> &n, const RCP<const Basic> &x)
+{
+    if ((is_a<Integer>(*n) and not (rcp_static_cast<const Integer>(n)->is_negative())) and
+        (is_a<Integer>(*x) and not (rcp_static_cast<const Integer>(x)->is_negative()))) {
```

Comment:
Should be just 
```
if ((is_a<Integer>(*n) and not (rcp_static_cast<const Integer>(n)->is_negative()) {
    return false;
}
```
If `n` is a non-negative integer, then it is reducible to the zeta form.
When it` is not, then it is not reducible (not confirmed)